polynote-kernel/src/main/scala/polynote/kernel/ScalaCompiler.scala

```diff
@@ -659,11 +659,13 @@ object ScalaCompiler {
     val cp = classPath ++ requiredPaths
 
     val settings = initial.copy()
-    settings.target.value = "jvm-1.8" // set Java8 by default
+    settings.target.value = if (settings.target.choices.contains("jvm-1.8")) "jvm-1.8" else "8" // set Java8 by default
     settings.classpath.append(cp.map(_.getCanonicalPath).mkString(File.pathSeparator))
     settings.Yrangepos.value = true
     try {
-      settings.YpartialUnification.value = true
+      Option(settings.getClass.getMethod("YpartialUnification")).foreach {
+        method => method.invoke(settings).asInstanceOf[settings.BooleanSetting].value = true
+      }
```

Comment:
Is this reflection-based thing needed for Scala 2.13?

Reply:
Yes, that option was dropped in 2.13 (it's always on) so you get a compile error without the reflection route (or splitting into 2.11/2.12/2.13 sources, which seemed like a lot in this case)

Reply:
Ah, gotcha.

Reply:
Added an explanatory comment here